src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.CompletionListUpdater.cs

```diff
@@ -58,6 +57,8 @@ private sealed class CompletionListUpdater
             // called concurrently, which essentially makes the pooled list a singleton,
             // but we still use ObjectPool for concurrency handling just to be robust.
             private static readonly ObjectPool<List<MatchResult<VSCompletionItem>>> s_listOfMatchResultPool = new(factory: () => new(), size: 1);
+            private static readonly ObjectPool<List<(RoslynCompletionItem, PatternMatch?)>> s_listOfItemMatchPairPool = new(factory: () => new(), size: 1);
+            private static readonly ObjectPool<ImmutableArray<RoslynCompletionItem>.Builder> s_filteredItemBuilderPool = new(factory: () => ImmutableArray.CreateBuilder<RoslynCompletionItem>(), size: 1);
```

Comment:
overall very confused about why we do a List in one place, but a IA.Builder in another.  Feels very strange.  Similarly, i'm personally surprised to see us capping the size at '1'.  in that, in practice the default cap feels like it's totally fine.  if we then have multiple, we still pool.  But if we only have one, it's still fine.

Reply:
>  us capping the size at '1'

I've done this because we absolutely don't expect this code path get executed in parallel (which would be a bug), and setting the size to 1 makes the expectation clear. But I agree that it should behave the same if everything works correctly (and avoid excessive allocations otherwise)